src/coreclr/vm/interopconverter.cpp

```diff
@@ -189,17 +189,26 @@ IUnknown *GetComIPFromObjectRef(OBJECTREF *poref, ComIpType ReqIpType, ComIpType
         {
             hr = SafeQueryInterface(pUnk, IID_IDispatch, &pvObj);
             pUnk->Release();
```

Comment:
This `Release` shouldn't be called indiscriminately here. We are checking for `IDispatch` now and it could fail, but `ComIpType_Unknown` must still be returned. Move this release into the success path below.